sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -516,6 +516,14 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("SPARK-48413: Alter column with collation") {
+    val tableName = "testcat.alter_column_tbl"
+    withTable(tableName) {
+      sql(s"CREATE TABLE $tableName (c1 STRING) USING PARQUET")
+      sql(s"ALTER TABLE $tableName ALTER COLUMN c1 TYPE STRING COLLATE UTF8_LCASE")
+    }
```

Comment:
Could you just for readability and completeness check that the collation of column c1 changed? Knowing that we passed the change, but not checking what we actually did is not sufficient.